Mathlib/MeasureTheory/Function/LpSpace.lean

```diff
@@ -788,6 +806,18 @@ theorem norm_indicatorConstLp_top (hμs_ne_zero : μ s ≠ 0) :
     ENNReal.rpow_zero, mul_one, ENNReal.coe_toReal, coe_nnnorm]
 #align measure_theory.norm_indicator_const_Lp_top MeasureTheory.norm_indicatorConstLp_top
 
+open scoped symmDiff in
+/-- Compute the `ℒᵖ` norm of the difference of two constant indicators of sets with finite measure
+and same constant using symmetric difference. -/
+theorem norm_indicatorConstLp_sub {t : Set α} (ht : MeasurableSet t) (hμt : μ t ≠ ∞)
+    (hp_ne_zero : p ≠ 0) (hp_ne_top : p ≠ ∞) :
+    ‖indicatorConstLp p hs hμs c - indicatorConstLp p ht hμt c‖ =
+    ‖c‖ * (μ (s ∆ t)).toReal ^ (1 / p.toReal) := by
+  rw [indicatorConstLp, indicatorConstLp, ← Memℒp.toLp_sub, Lp.norm_toLp, ← snorm_norm]
+  simp_rw [Pi.sub_apply, ← Set.apply_indicator_symmDiff norm_neg]
+  rw [snorm_norm, snorm_indicator_const (hs.symmDiff ht) hp_ne_zero hp_ne_top, ENNReal.toReal_mul,
+    toReal_coe_nnnorm, ← ENNReal.toReal_rpow]
```

Comment:
Since you said you would prefer a single sequence of rewrite, here's how I would do it:
```suggestion
  simp_rw [indicatorConstLp, ← Memℒp.toLp_sub, Lp.norm_toLp, ← snorm_norm (F := E),
    Pi.sub_apply, ← apply_indicator_symmDiff norm_neg, snorm_norm, 
    snorm_indicator_const (hs.symmDiff ht) hp_ne_zero hp_ne_top, ENNReal.toReal_mul,
    toReal_coe_nnnorm, ← ENNReal.toReal_rpow]
```

Note the key trick to force `F := E` to prevent an infinite loop when rewriting by `← snorm_norm`.